src/Symfony/Component/String/UnicodeString.php

```diff
@@ -332,7 +336,15 @@ public function startsWith($prefix): bool
         $form = null === $this->ignoreCase ? \Normalizer::NFD : \Normalizer::NFC;
         normalizer_is_normalized($prefix, $form) ?: $prefix = normalizer_normalize($prefix, $form);
 
-        return '' !== $prefix && false !== $prefix && 0 === ($this->ignoreCase ? grapheme_stripos($this->string, $prefix) : grapheme_strpos($this->string, $prefix));
+        if ('' === $prefix || false === $prefix) {
+            return false;
+        }
+
+        if ($this->ignoreCase) {
+            return 0 === grapheme_stripos(grapheme_extract($this->string, \strlen($prefix), GRAPHEME_EXTR_MAXBYTES), $prefix);
```

Comment:
hmm, maybe we could even use `mb_stripos` instead of `grapheme_stripos`, because grapheme_extract ensures that we do not get half grapheme clusters. Or is there a test case that would fail with `mb_stripos`?

Reply:
the existing test cases are succeeding.